internal/dataplane/translator/translate_certs.go

```diff
@@ -72,14 +79,19 @@ func (t *Translator) getGatewayCerts() []certWrapper {
 				continue
 			}
 
-			// Check if listener is marked as programmed
-			if !util.CheckCondition(
-				status.Conditions,
-				util.ConditionType(gatewayapi.ListenerConditionProgrammed),
-				util.ConditionReason(gatewayapi.ListenerReasonProgrammed),
-				metav1.ConditionTrue,
-				gateway.Generation,
-			) {
+			// Check if listener is marked as programmed when the gateway is controlled by KIC  and corresponding GatewayClass has the "Unmanaged" annotation.
+			// If the GatewayClass does not satisfy the condition, the gateway is considered to be managed by other components (for example Kong Operator),
+			// so we do not check the "Programmed" condition before extracting the certificate from the listener
+			// to prevent unexpected deletion of certificates when the instancec is managed by Kong Operator.
+			if gwc.Spec.ControllerName == gatewayapi.GatewayController(t.gatewayControllerName) &&
```

Comment:
Is this logic correct here? 🤔 

```
gwc.Spec.ControllerName == gatewayapi.GatewayController(t.gatewayControllerName)
```

So if the controller name is different then we perform the listener tls logic below which I'm not sure is something we'd like, right? (in that case it's not the  `Gateway` which we want to reconcile)

I _think_ that we want to move this check above, just below getting the `gwc`, so that:

```
if gwc.Spec.ControllerName != gatewayapi.GatewayController(t.gatewayControllerName) {
  continue
}
```

WDYT?

Reply:
I think when the controller name is different, the gateway will not we added to the translator cache. So do you think we should remove the controller name check?